src/Microsoft.TemplateEngine.Abstractions/IGenerator.cs

```diff
@@ -57,6 +58,29 @@ public interface IGenerator : IIdentifiedComponent
         /// <returns><see cref="IParameterSet"/> with parameters available in <paramref name="template"/>.</returns>
         IParameterSet GetParametersForTemplate(IEngineEnvironmentSettings environmentSettings, ITemplate template);
 
+        /// <summary>
+        /// Evaluates possible conditions defined on parameters within the given set of parameters.
+        /// Performs the changes on the given set of parameters as needed and returns the summary of changes performed.
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <param name="parameters">Parameters to be evaluated and updated.</param>
+        /// <param name="template"></param>
+        /// <returns></returns>
+        ParametersConditionsEvaluationResult EvaluateConditionalParameters(ILogger logger, IParameterSet parameters, ITemplate template);
```

Comment:
Should we return `IParameterSet` here with evaluated conditions instead of new type? 
I.e. each parameter will have IsRequired/IsEnabled with states `true`, `false`.  `not evaluated` may be considered as `false`.